### PR TITLE
test to cover parallel execution and networking

### PR DESCRIPTION
**Pull request by muayyad-alsadi**
creates 10 containers in parallel having no image (podman rmi)
removes them in parallel

do the same again but having the image existing before we start

this opens host port 8080, do some test the close it, if you want to customize this please tell me.

sorry I closed #2552 by deleting the previous branch mistake while doing a rebase.
some details are in #2551

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM

File: test/test_podman_baseline.sh [hunk 1]
```diff
@@ -78,7 +78,71 @@ echo $image
...
+echo "$txt" > /tmp/hello.txt
+podman run -d --name myweb -p 8080:80 -w /var/www -v /tmp/hello.txt:/var/www/index.txt busybox httpd -f -p 80
+echo $txt | podman exec -i myweb sh -c "cat > /var/www/index2.txt"
+txt2=$( podman exec myweb cat /var/www/index2.txt )
+[ "x$txt1"=="x$txt2" ] && echo "PASS"
+txt2=$( podman run --rm --net host busybox wget -qO - http://localhost:8080/index.txt )
+[ "x$txt1"=="x$txt2" ] && echo "PASS"
+txt2=$( podman run --rm --net host busybox wget -qO - http://localhost:8080/index2.txt )
+[ "x$txt1"=="x$txt2" ] && echo "PASS"
+# poadman run --rm --net container:myweb --add-host myweb:127.0.0.1 busybox wget -qO - http://myweb/index.txt
```
Comment (TomSweeneyRedHat) [inline test/test_podman_baseline.sh]: Should be podman, not podman and unless you plan to use it at some point, I'd just remove it.

File: test/test_podman_baseline.sh [hunk 2]
```diff
@@ -78,7 +78,71 @@ echo $image
...
+########
+txt1="Hello, Podman"
+echo "$txt" > /tmp/hello.txt
+podman run -d --name myweb -p 8080:80 -w /var/www -v /tmp/hello.txt:/var/www/index.txt busybox httpd -f -p 80
+echo $txt | podman exec -i myweb sh -c "cat > /var/www/index2.txt"
+txt2=$( podman exec myweb cat /var/www/index2.txt )
+[ "x$txt1"=="x$txt2" ] && echo "PASS"
+txt2=$( podman run --rm --net host busybox wget -qO - http://localhost:8080/index.txt )
+[ "x$txt1"=="x$txt2" ] && echo "PASS"
+txt2=$( podman run --rm --net host busybox wget -qO - http://localhost:8080/index2.txt )
```
Comment (TomSweeneyRedHat) [inline test/test_podman_baseline.sh]: needs `run -d`

File: test/test_podman_baseline.sh [hunk 3]
```diff
@@ -78,7 +78,71 @@ echo $image
...
+########
+# Test networking, bind mounting a file, stdin/stdout redirect
+########
+txt1="Hello, Podman"
+echo "$txt" > /tmp/hello.txt
+podman run -d --name myweb -p 8080:80 -w /var/www -v /tmp/hello.txt:/var/www/index.txt busybox httpd -f -p 80
+echo $txt | podman exec -i myweb sh -c "cat > /var/www/index2.txt"
+txt2=$( podman exec myweb cat /var/www/index2.txt )
+[ "x$txt1"=="x$txt2" ] && echo "PASS"
+txt2=$( podman run --rm --net host busybox wget -qO - http://localhost:8080/index.txt )
```
Comment (TomSweeneyRedHat) [inline test/test_podman_baseline.sh]: needs `run -d`

File: test/test_podman_baseline.sh [hunk 4]
```diff
@@ -78,7 +78,71 @@ echo $image
...
+count=$( podman ps -q  | wc -l )
+[ "x$count" == "x10" ] || { echo "was expecting 10 running containers"; exit -1; }
+
+for i in `seq 10`; do ( podman stop -t=1 c$i; podman rm c$i )& done
+echo -e "\nwaiting for deletion...\n"
+wait
+echo -e "\ndone\n"
+# assert we have 0 running containers
+count=$( podman ps -q  | wc -l )
+[ "x$count" == "x0" ] || { echo "was expecting 0 running containers"; exit -1; }
```
Comment (TomSweeneyRedHat) [inline test/test_podman_baseline.sh]: I forgot to mention/ask for this yesterday, my apologies.  I like this test to not stop if there's an error unless the '-e' param is passed in.  For this line and the others where you exit on error, can you rework it slightly  to something like:

```
[ "x$count" == "x0" ] || { echo "was expecting 0 running containers";-}
if [ "$showerror" eq 1 ]; then
   [ "x$count" == "x0" ] || { exit -1; }
fi
```

**Comment by TomSweeneyRedHat** [comment]
Really nice addition to the test.  Couple of comments and then gtg.

File: test/test_podman_baseline.sh (hunk 1 above)
Comment (muayyad-alsadi) [reply test/test_podman_baseline.sh]: yes, I'll uncomment it soon when it works, see my comment here. @haircommander fixed it but I did not yet validate it

https://github.com/containers/libpod/issues/2504#issuecomment-469764333

File: test/test_podman_baseline.sh (hunk 2 above)
Comment (muayyad-alsadi) [reply test/test_podman_baseline.sh]: no it does not. I'm not taking the container id, I'm testing the output of `wget -qO -` , passing -d would make it go background and no output is stored in `txt2`

File: test/test_podman_baseline.sh (hunk 3 above)
Comment (muayyad-alsadi) [reply test/test_podman_baseline.sh]: it does not, you can try it.

File: test/test_podman_baseline.sh (hunk 4 above)
Comment (muayyad-alsadi) [reply test/test_podman_baseline.sh]: I'll fix that

File: test/test_podman_baseline.sh (hunk 2 above)
Comment (TomSweeneyRedHat) [reply test/test_podman_baseline.sh]: When I ran it, the test hung there for me, that's why I thought it needed the `-d`...

File: test/test_podman_baseline.sh (hunk 3 above)
Comment (muayyad-alsadi) [reply test/test_podman_baseline.sh]: I find some typos I'll fix them.

File: test/test_podman_baseline.sh (hunk 2 above)
Comment (muayyad-alsadi) [reply test/test_podman_baseline.sh]: please try now

File: test/test_podman_baseline.sh [hunk 5]
```diff
@@ -83,24 +84,32 @@ podman run --rm $image ls -alF /etc
...
-[ "x$txt1"=="x$txt2" ] && echo "PASS"
-txt2=$( podman run --rm --net host busybox wget -qO - http://localhost:8080/index2.txt )
-[ "x$txt1"=="x$txt2" ] && echo "PASS"
-# poadman run --rm --net container:myweb --add-host myweb:127.0.0.1 busybox wget -qO - http://myweb/index.txt
+[ "x$txt1" == "x$txt2" ] && echo "PASS1" || { echo "FAIL1"; port_test_failed=1; }
+txt2=$( podman run --rm --net host busybox wget -qO - http://localhost:$HOST_PORT/index.txt )
+[ "x$txt1" == "x$txt2" ] && echo "PASS2" || { echo "FAIL2"; port_test_failed=1; }
+txt2=$( podman run --rm --net host busybox wget -qO - http://localhost:$HOST_PORT/index2.txt )
+[ "x$txt1" == "x$txt2" ] && echo "PASS3" || { echo "FAIL3"; port_test_failed=1; }
+# podman run --rm --net container:myweb --add-host myweb:127.0.0.1 busybox wget -qO - http://myweb/index.txt
```
Comment (muayyad-alsadi) [inline test/test_podman_baseline.sh]: if the @haircommander  fix worked, we can have

``` txt2=$( podman run --rm --net container:myweb --add-host myweb:127.0.0.1 busybox wget -qO - http://myweb/index.txt )
 [ "x$txt1" == "x$txt2" ] && echo "PASS4" || { echo "FAIL4"; port_test_failed=1; }
```

File: test/test_podman_baseline.sh [hunk 6]
```diff
@@ -78,7 +79,99 @@ echo $image
...
+
+for i in `seq 10`; do ( podman stop -t=1 c$i; podman rm c$i )& done
+echo -e "\nwaiting for deletion...\n"
+wait
+echo -e "\ndone\n"
+# assert we have 0 running containers
+count=$( podman ps -q  | wc -l )
+[ "x$count" == "x0" ] && echo "PASS" || { echo "FAIL, expecting 0 found $count"; prun_test_failed=1; }
+[ "0$prun_test_failed" -eq 1 ] && [ "0$showerror" -eq 1 ] && {
+  echo "was expecting 0 running containers";
```
Comment (TomSweeneyRedHat) [inline test/test_podman_baseline.sh]: @muayyad-alsadi it's much happier now.  Thanks for the changes.
If you've the time,can you have each of the echo's for errors like this line print if you had the failure?  If running it all but not stopping on errors, I still would like to see the errors.

If not, no worries.  I'd a thought today to add color to the error lines.  I can make the change myself when I go through that exercise.

**Comment by TomSweeneyRedHat** [comment]
One hopefully small question, otherwise LGTM.

File: test/test_podman_baseline.sh (hunk 6 above)
Comment (muayyad-alsadi) [reply test/test_podman_baseline.sh]: Let's add those when I revisit it to add the commented out tests about `--net container:`

**Comment by TomSweeneyRedHat** [comment]
OK, LGTM then.

**Pull request merged**